Index: ZipFile.cs
===================================================================
--- ZipFile.cs	(revision 524)
+++ ZipFile.cs	(working copy)
@@ -1257,7 +1257,7 @@
 					}
 
 					if (compressedSize != entry.CompressedSize &&
-						compressedSize != 0xFFFFFFFF && compressedSize != -1) {
+						compressedSize != 0xFFFFFFFF && compressedSize != -1 && compressedSize != 0) {
 						throw new ZipException(
 							string.Format("Compressed size mismatch between central header({0}) and local header({1})",
 							entry.CompressedSize, compressedSize));
