Index: ZipEntry.cs
===================================================================
--- ZipEntry.cs	(revision 524)
+++ ZipEntry.cs	(working copy)
@@ -1004,7 +1004,7 @@
 							long lastAccess = extraData.ReadLong();
 							long createTime = extraData.ReadLong();
 
-							DateTime = System.DateTime.FromFileTime(lastModification);
+							DateTime = System.DateTime.FromFileTimeUtc(lastModification);
 						}
 						break;
 					}
@@ -1023,8 +1023,8 @@
 				if ( ((flags & 1) != 0) && (length >= 5) ) {
 					int iTime = extraData.ReadInt();
 
-					DateTime = (new System.DateTime ( 1970, 1, 1, 0, 0, 0 ).ToUniversalTime() +
-						new TimeSpan ( 0, 0, 0, iTime, 0 )).ToLocalTime();
+					DateTime = (new System.DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc) +
+						new TimeSpan(0, 0, 0, iTime, 0));
 				}
 			}
 			if (method == CompressionMethod.WinZipAES) {
Index: ZipExtraData.cs
===================================================================
--- ZipExtraData.cs	(revision 524)
+++ ZipExtraData.cs	(working copy)
@@ -197,24 +197,24 @@
 				{
 					int iTime = helperStream.ReadLEInt();
 
-					_modificationTime = (new DateTime(1970, 1, 1, 0, 0, 0).ToUniversalTime() +
-						new TimeSpan(0, 0, 0, iTime, 0)).ToLocalTime();
+					_modificationTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc) +
+						new TimeSpan(0, 0, 0, iTime, 0);
 				}
 
 				if ((_flags & Flags.AccessTime) != 0)
 				{
 					int iTime = helperStream.ReadLEInt();
 
-					_lastAccessTime = (new DateTime(1970, 1, 1, 0, 0, 0).ToUniversalTime() +
-						new TimeSpan(0, 0, 0, iTime, 0)).ToLocalTime();
+					_lastAccessTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc) +
+						new TimeSpan(0, 0, 0, iTime, 0);
 				}
 				
 				if ((_flags & Flags.CreateTime) != 0)
 				{
 					int iTime = helperStream.ReadLEInt();
 
-					_createTime = (new DateTime(1970, 1, 1, 0, 0, 0).ToUniversalTime() +
-						new TimeSpan(0, 0, 0, iTime, 0)).ToLocalTime();
+					_createTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc) +
+						new TimeSpan(0, 0, 0, iTime, 0);
 				}
 			}
 		}
@@ -231,17 +231,17 @@
 				helperStream.IsStreamOwner = false;
 				helperStream.WriteByte((byte)_flags);     // Flags
 				if ( (_flags & Flags.ModificationTime) != 0) {
-					TimeSpan span = _modificationTime.ToUniversalTime() - new DateTime(1970, 1, 1, 0, 0, 0).ToUniversalTime();
+					TimeSpan span = _modificationTime - new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
 					int seconds = (int)span.TotalSeconds;
 					helperStream.WriteLEInt(seconds);
 				}
 				if ( (_flags & Flags.AccessTime) != 0) {
-					TimeSpan span = _lastAccessTime.ToUniversalTime() - new DateTime(1970, 1, 1, 0, 0, 0).ToUniversalTime();
+					TimeSpan span = _lastAccessTime - new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
 					int seconds = (int)span.TotalSeconds;
 					helperStream.WriteLEInt(seconds);
 				}
 				if ( (_flags & Flags.CreateTime) != 0) {
-					TimeSpan span = _createTime.ToUniversalTime() - new DateTime(1970, 1, 1, 0, 0, 0).ToUniversalTime();
+					TimeSpan span = _createTime - new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
 					int seconds = (int)span.TotalSeconds;
 					helperStream.WriteLEInt(seconds);
 				}
