Index: src/Zip/ZipEntry.cs
===================================================================
--- src/Zip/ZipEntry.cs	(revision 524)
+++ src/Zip/ZipEntry.cs	(working copy)
@@ -1004,7 +1004,7 @@
 							long lastAccess = extraData.ReadLong();
 							long createTime = extraData.ReadLong();
 
-							DateTime = System.DateTime.FromFileTime(lastModification);
+							DateTime = System.DateTime.FromFileTimeUtc(lastModification);
 						}
 						break;
 					}
@@ -1023,8 +1023,8 @@
 				if ( ((flags & 1) != 0) && (length >= 5) ) {
 					int iTime = extraData.ReadInt();
 
-					DateTime = (new System.DateTime ( 1970, 1, 1, 0, 0, 0 ).ToUniversalTime() +
-						new TimeSpan ( 0, 0, 0, iTime, 0 )).ToLocalTime();
+					DateTime = (new System.DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc) +
+						new TimeSpan(0, 0, 0, iTime, 0));
 				}
 			}
 			if (method == CompressionMethod.WinZipAES) {
Index: src/Zip/ZipExtraData.cs
===================================================================
--- src/Zip/ZipExtraData.cs	(revision 524)
+++ src/Zip/ZipExtraData.cs	(working copy)
@@ -197,24 +197,24 @@
 				{
 					int iTime = helperStream.ReadLEInt();
 
-					_modificationTime = (new DateTime(1970, 1, 1, 0, 0, 0).ToUniversalTime() +
-						new TimeSpan(0, 0, 0, iTime, 0)).ToLocalTime();
+					_modificationTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc) +
+						new TimeSpan(0, 0, 0, iTime, 0);
 				}
 
 				if ((_flags & Flags.AccessTime) != 0)
 				{
 					int iTime = helperStream.ReadLEInt();
 
-					_lastAccessTime = (new DateTime(1970, 1, 1, 0, 0, 0).ToUniversalTime() +
-						new TimeSpan(0, 0, 0, iTime, 0)).ToLocalTime();
+					_lastAccessTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc) +
+						new TimeSpan(0, 0, 0, iTime, 0);
 				}
 				
 				if ((_flags & Flags.CreateTime) != 0)
 				{
 					int iTime = helperStream.ReadLEInt();
 
-					_createTime = (new DateTime(1970, 1, 1, 0, 0, 0).ToUniversalTime() +
-						new TimeSpan(0, 0, 0, iTime, 0)).ToLocalTime();
+					_createTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc) +
+						new TimeSpan(0, 0, 0, iTime, 0);
 				}
 			}
 		}
@@ -231,17 +231,17 @@
 				helperStream.IsStreamOwner = false;
 				helperStream.WriteByte((byte)_flags);     // Flags
 				if ( (_flags & Flags.ModificationTime) != 0) {
-					TimeSpan span = _modificationTime.ToUniversalTime() - new DateTime(1970, 1, 1, 0, 0, 0).ToUniversalTime();
+					TimeSpan span = _modificationTime - new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
 					int seconds = (int)span.TotalSeconds;
 					helperStream.WriteLEInt(seconds);
 				}
 				if ( (_flags & Flags.AccessTime) != 0) {
-					TimeSpan span = _lastAccessTime.ToUniversalTime() - new DateTime(1970, 1, 1, 0, 0, 0).ToUniversalTime();
+					TimeSpan span = _lastAccessTime - new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
 					int seconds = (int)span.TotalSeconds;
 					helperStream.WriteLEInt(seconds);
 				}
 				if ( (_flags & Flags.CreateTime) != 0) {
-					TimeSpan span = _createTime.ToUniversalTime() - new DateTime(1970, 1, 1, 0, 0, 0).ToUniversalTime();
+					TimeSpan span = _createTime - new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
 					int seconds = (int)span.TotalSeconds;
 					helperStream.WriteLEInt(seconds);
 				}
@@ -374,13 +374,13 @@
 						if (ntfsLength >= 24)
 						{
 							long lastModificationTicks = helperStream.ReadLELong();
-							_lastModificationTime = DateTime.FromFileTime(lastModificationTicks);
+							_lastModificationTime = DateTime.FromFileTimeUtc(lastModificationTicks);
 
 							long lastAccessTicks = helperStream.ReadLELong();
-							_lastAccessTime = DateTime.FromFileTime(lastAccessTicks);
+                            _lastAccessTime = DateTime.FromFileTimeUtc(lastAccessTicks);
 
 							long createTimeTicks = helperStream.ReadLELong();
-							_createTime = DateTime.FromFileTime(createTimeTicks);
+                            _createTime = DateTime.FromFileTimeUtc(createTimeTicks);
 						}
 						break;
 					}
@@ -406,9 +406,9 @@
 				helperStream.WriteLEInt(0);       // Reserved
 				helperStream.WriteLEShort(1);     // Tag
 				helperStream.WriteLEShort(24);    // Length = 3 x 8.
-				helperStream.WriteLELong(_lastModificationTime.ToFileTime());
-				helperStream.WriteLELong(_lastAccessTime.ToFileTime());
-				helperStream.WriteLELong(_createTime.ToFileTime());
+				helperStream.WriteLELong(_lastModificationTime.ToFileTimeUtc());
+				helperStream.WriteLELong(_lastAccessTime.ToFileTimeUtc());
+                helperStream.WriteLELong(_createTime.ToFileTimeUtc());
 				return ms.ToArray();
 			}
 		}
@@ -482,9 +482,9 @@
 		}
 
 		#region Instance Fields
-		DateTime _lastAccessTime = DateTime.FromFileTime(0);
-		DateTime _lastModificationTime = DateTime.FromFileTime(0);
-		DateTime _createTime = DateTime.FromFileTime(0);
+        DateTime _lastAccessTime = DateTime.FromFileTimeUtc(0);
+        DateTime _lastModificationTime = DateTime.FromFileTimeUtc(0);
+        DateTime _createTime = DateTime.FromFileTimeUtc(0);
 		#endregion
 	}
 
