Index: ZipFile.cs
===================================================================
--- src/Zip/ZipFile.cs	(revision 524)
+++ src/Zip/ZipFile.cs	(working copy)
@@ -3304,7 +3304,7 @@
 					ZipAESTransform decryptor = new ZipAESTransform(rawPassword_, saltBytes, blockSize, false);
 					byte[] pwdVerifyCalc = decryptor.PwdVerifier;
 					if (pwdVerifyCalc[0] != pwdVerifyRead[0] || pwdVerifyCalc[1] != pwdVerifyRead[1])
-						throw new Exception("Invalid password for AES");
+						throw new ZipException("Invalid password");
 					result = new ZipAESStream(baseStream, decryptor, CryptoStreamMode.Read);
 				}
 				else
