Index: TarInputStream.cs
===================================================================
--- TarInputStream.cs	(revision 524)
+++ TarInputStream.cs	(working copy)
@@ -493,6 +493,7 @@
 						headerBuf = this.tarBuffer.ReadBlock();
 					} else if (header.TypeFlag != TarHeader.LF_NORMAL && 
 							   header.TypeFlag != TarHeader.LF_OLDNORM &&
+                               header.TypeFlag != TarHeader.LF_SYMLINK &&
 							   header.TypeFlag != TarHeader.LF_DIR) {
 						// Ignore things we dont understand completely for now
 						SkipToNextEntry();
