# Bastian Eicher <bastian@eicher.net>
Sparse arguments

diff -r 904899a9cead -r b4a8d158afc5 src/Options.cs
--- a/src/Options.cs	Tue Jun 25 15:38:08 2013 +0200
+++ b/src/Options.cs	Tue Jun 25 15:17:22 2013 +0200
@@ -709,7 +709,7 @@
 				from argument in arguments
 				where ++c.OptionIndex >= 0 && (process || def != null)
 					? process
-						? argument == "--" 
+						? argument == "--" && Count > 0
 							? (process = false)
 							: !Parse (argument, c)
 								? def != null 
@@ -736,7 +736,7 @@
 			Option def = Contains ("<>") ? this ["<>"] : null;
 			foreach (string argument in arguments) {
 				++c.OptionIndex;
-				if (argument == "--") {
+				if (argument == "--" && Count > 0) {
 					process = false;
 					continue;
 				}
